src/App.js

```diff
@@ -1,23 +1,69 @@
-import logo from './logo.svg';
-import './App.css';
+import { useState } from 'react';
+import PageHeader from './components/PageHeader';
+import PageMain from './components/PageMain';
+import PageFooter from './components/PageFooter';
+import SearchForm from './components/SearchForm';
+import MovieDetails from './components/MovieDetails';
+import MoviesList from './components/MoviesList';
+import MoviesModal from './components/MoviesModal';
+import ErrorBoundary from './utils/ErrorBoundary';
+import './styles/App.scss';
 
 function App() {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [isMovieDetailsMode, setIsMovieDetailsMode] = useState(false);
+  const [movieDetails, setMovieDetails] = useState({ type: 'add', data: {} });
+
+  const showMovieModal = (type, movieData) => {
+    setMovieDetails({ type, data: movieData });
+    setIsModalVisible(true);
+  };
+
+  const toggleMovieDetailsMode = () => {
+    return isMovieDetailsMode
+      ? setIsMovieDetailsMode(false)
+      : showMovieModal('add');
+  };
+
+  const movieClickHandler = movie => {
```

Comment:
movieClickHandler -> showMovieDetails (because movieClickHandler says nothing about what it will do and other devs and you in the future will always have to go to this fn and read it line by line to understand what is happening)